cli/api/logs.go

```diff
@@ -49,10 +83,103 @@ type ExportLogsConfig struct {
 //
 // TODO: This code is racy - creating then erasing the output file does not
 // guarantee that it will be safe to write to at the end of the function
-func (a *api) ExportLogs(config ExportLogsConfig) (err error) {
+func (a *api) ExportLogs(configParam ExportLogsConfig) (err error) {
 	var e error
-	files := []*os.File{}
-	fileIndex := make(map[string]map[string]int) // host => filename => index
+	e = validateConfiguration(&configParam)
+	if e != nil {
+		return e
+	}
+
+	var exporter *logExporter
+	defer func () {
+		if exporter != nil {
+			exporter.cleanup()
+		}
+	}()
+
+	exporter, e = buildExporter(configParam, a.GetServices, a.GetHostMap)
+	if e != nil {
+		return e
+	}
+
+	numWarnings := 0
+	foundIndexedDay := false
+
+	glog.Infof("Starting part 1 of 3: process logstash elasticsearch results using temporary dir: %s", exporter.tempdir)
+	foundIndexedDay, numWarnings, e = exporter.retrieveLogs()
+	if e != nil {
+		return e
+	} else if !foundIndexedDay {
+		return fmt.Errorf("no logstash indexes exist for the given date range %s - %s", exporter.FromDate, exporter.ToDate)
+	}
+
+	glog.Infof("Starting part 2 of 3: sort %d output files", len(exporter.outputFiles))
+
+	indexData := []string{}
+	for i, outputFile := range exporter.outputFiles {
+		filename := filepath.Join(exporter.tempdir, fmt.Sprintf("%03d.log", i))
+		tmpfilename := filepath.Join(exporter.tempdir, fmt.Sprintf("%03d.log.tmp", i))
+		cmd := exec.Command("sort", filename, "-uo", tmpfilename)
```

Comment:
I'm just reusing the cards I was dealt.  
What are the GO utilities for sort file and copy file?
I did find `os.Rename` to replace `mv`, but I didn't find one-line equivalents for `cp` and `sort`